amlb/runners/docker.py

```diff
@@ -51,9 +51,22 @@ def _start_container(self, script_params=""):
             touch(d, as_dir=True)
         script_extra_params = ""
         inst_name = self.sid
+
+        if ' -f ' in script_params:
+             inst_name = inst_name + '_f' + script_params.split(' -f ')[1].split()[0]
+        elif ' --fold ' in script_params:
+            inst_name = inst_name + '_f' + script_params.split(' --fold ')[1].split()[0]
+
+        memory = ""
+        if rconfig().docker.constraints_enabled:
+            if "max_mem_size_mb" in self.constraint_def:
+                memory = "--memory={}m".format(self.constraint_def['max_mem_size_mb'])
+                memory += " " + "--memory-swap={}".format(self.constraint_def['swap'])
```

Comment:
if you plan to add `max_mem_size_mb` property to constraints, then it should be supported by all runners, not only docker.

> constraint_def['swap']

swap logic doesn't make sense in constraint definition imo, it can be set globally.